src/FilterEngine.cpp

```diff
@@ -246,19 +280,16 @@ void FilterEngine::CreateAsync(const JsEnginePtr& jsEngine,
       filterEngine->GetJsEngine().NotifyLowMemory();
   });
 
-  // Lock the JS engine while we are loading scripts, no timeouts should fire
-  // until we are done.
-  const JsContext context(*jsEngine);
```

Comment:
I've moved "const JsContext context(*jsEngine);" to the implementation of EvaluateCallback and it is called before JsEngine::Evaluate() (actually I call it too early as I noted now and it could be called past the "if (jsSources[i] == filename)" condition).
Do you want to remove it from the callback and make the caller to acquire this lock?

Reply:
Nevermind my previous comment. I missed that tou wrote "between" :-).